product_supplierinfo_for_customer_picking/models/stock_move.py

```diff
@@ -1,30 +1,33 @@
-# Copyright 2013-2017 Agile Business Group sagl
-#     (<http://www.agilebg.com>)
+# Copyright 2013 - 2021 Agile Business Group sagl (<https://www.agilebg.com>)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 from odoo import api, fields, models
 
 
 class StockMove(models.Model):
     _inherit = "stock.move"
 
-    @api.multi
     @api.depends(
         "picking_id.partner_id", "product_id", "product_id.customer_ids.product_code"
     )
     def _compute_product_customer_code(self):
-        for move in self.filtered(
-            lambda m: m.picking_id
-            and m.picking_id.partner_id
-            and m.product_tmpl_id.customer_ids
-        ):
-            customers = move.product_tmpl_id.customer_ids.filtered(
-                lambda m: move.picking_id.partner_id
-            )
-            if customers:
-                move.product_customer_code = customers[0].product_code
+        for move in self:
+            if (
+                not move.picking_id
+                or not move.picking_id.partner_id
+                or not move.product_tmpl_id.customer_ids
+            ):
+                move.product_customer_code = False
+                continue
+            else:
+                customers = move.product_tmpl_id.customer_ids.filtered(
+                    lambda m: move.picking_id.partner_id
+                )
+                product_customer_code = (
+                    customers[0].product_code if customers else False
+                )
+                move.product_customer_code = product_customer_code
```

Comment:
you can probably perform refactoring here with something like:

          ```
          product_customer_code = False
            if (
                move.picking_id
                and move.picking_id.partner_id
                and move.product_tmpl_id.customer_ids
            ):
                customer = fields.first(move.product_tmpl_id.customer_ids.filtered(
                    lambda m: move.picking_id.partner_id
                ))
                product_customer_code = customer.product_code
            move.product_customer_code = product_customer_code
```

what do you think?

Reply:
Definitely better! Done, thanks.